base/float.jl

```diff
@@ -386,10 +386,35 @@ end
 end
 
 # fused multiply-add
-fma_libm(x::Float32, y::Float32, z::Float32) =
-    ccall(("fmaf", libm_name), Float32, (Float32,Float32,Float32), x, y, z)
-fma_libm(x::Float64, y::Float64, z::Float64) =
-    ccall(("fma", libm_name), Float64, (Float64,Float64,Float64), x, y, z)
+function fma_libm(x::Float32, y::Float32, z::Float32)
+    r = ccall(("fmaf", libm_name), Float32, (Float32,Float32,Float32), x, y, z)
+    # This is necessary on 32-bit Intel Linux, probably on OS X, and
+    # maybe on Windows
+    # TODO: Disable this for non-Intel systems
+    if is(Int, Int32)
```

Comment:
> const WORD_SIZE = int(Int.size)*8

Yeah, that's the same, I was just making a stylistic point.

> I think I should actually be looking at Base.ARCH instead. Do you know what value it has for a 32-bit Intel system? :i686?

`ARCH` is passed to gcc/clang, so it may be set to any value that -march/mtune support. For 32-bit Intel, this is a very long list which evolves over time: https://gcc.gnu.org/onlinedocs/gcc-4.5.3/gcc/i386-and-x86_002d64-Options.html So it's not possible to future-proof detection if somebody sets `ARCH = some_new_cpu` later in a build.

So it looks like we'd better use compiler macros instead: http://sourceforge.net/p/predef/wiki/Architectures/ or http://nadeausoftware.com/articles/2012/02/c_c_tip_how_detect_processor_type_using_compiler_predefined_macros Adding a global variable containing an enum to `base.build_h.jl` looks like the best solution, so that it could be used to detect e.g. ARM if needed.